### ISSUE TITLE
Client performance setting presets for low-/mid-/high- end PCs

### ISSUE THREAD

**Issue by MarkuBu** [luanti-org/luanti#5024]
The default settings in Minetest are pretty conservative. If someone has a gaming PC he has to study the minetest.conf for better settings, but it is difficult to find all parameters.

I suggest to add some presets for values like these

```
client_unload_unused_data_timeout = 800
client_mapblock_limit = 10000
active_object_send_range_blocks = 5
active_block_range = 5
max_block_send_distance = 15
max_block_generate_distance = 19
mip_map = true
anisotropic_filter = true
enable_waving_leaves = true
enable_waving_plants = true
…
```

3 to 4 presets for a view range of 70 or below for low end PCs, 100, 200 and maybe 300 or more for high end gaming PCs

I have a mid range PC with Nvidia GTX750 and AMD FX-6300 and I play with these settings and a view range of 250 and it makes a huge difference 

view range 250
![screenshot_20170112_084541](https://cloud.githubusercontent.com/assets/16163058/21881297/0019da24-d8a4-11e6-90c2-d2537e97afac.jpg)

view range 100
![screenshot_20170112_084604](https://cloud.githubusercontent.com/assets/16163058/21881321/0c0fb736-d8a4-11e6-9682-777c6c493c21.jpg)

view range 250
![screenshot_20170112_085421](https://cloud.githubusercontent.com/assets/16163058/21881491/dd21b7fc-d8a4-11e6-87c5-564c94bd2f16.jpg)

view range 100
![screenshot_20170112_085404](https://cloud.githubusercontent.com/assets/16163058/21881503/ec1ae256-d8a4-11e6-8ec7-2288b247d63f.jpg)

But it is difficult for unexperienced users to find good settings

**Comment by sofar** [comment]
Honestly, the view range should auto adjust based on desired frame rate: If you prefer 30fps, and it has enough cycles to draw more map, it should slightly increase view range automatically. If you prefer 100fps and the game is short on time to draw the selected view range, it should reduce the view range a bit.

**Comment by MarkuBu** [comment]
But to profit from higher view ranges you may need higher values for e.g. max_block_generate_distance, because can see further than the map is generated. And other settings might also effect the performance, e.g. if you have 8 or 16GB of Ram and you can store more map blocks.

It is also strange if mobs suddenly appear/disappear because their active range is lower than the view range. 

And auto adjust view range was removed in 0.1.14, afaik

**Comment by beyondlimits** [comment]
You can use + and - keys to change visible range at runtime.

**Comment by MarkuBu** [comment]
I know that, but as explained it doesn't makes sense to just adjust view range if the map is not generated and mobs just disappear within view range.

Also it might me necessary to modify other parameters to play with high view ranges

**Comment by paramat** [comment]
> And auto adjust view range was removed

Yes for good reason, we kept fiddling with it to make it work well but it never did, it was either too slow or bounced around like crazy reacting to load. Much better how it is now being manual.

Maybe to be clearer you could change the thread title to refer to multiple settings, not just view range?

**Comment by MarkuBu** [comment]
Maybe you're right, the title is not clear. But I think, it should be clear now what I mean?

**Comment by MarkuBu** [comment]
I changed the title. Hope it is a little bit more clear now. But the word "preset" should be clear enough.

I made a little mockup

![bildschirmfoto vom 2017-01-13 13-54-42](https://cloud.githubusercontent.com/assets/16163058/21931207/855dd734-d99a-11e6-9cf6-2dd1e25e8a7e.png)

And I try to explain it, hopefully. With this settings you not only set the default view range. You also change the settings for all parameters in minetest.conf that are relevant

For example: the current default setting is max_block_generate_distance = 6 and max_block_send_distance = 10. This is ok for a view range of 100, but not for e.g. 250. Here are two screenshots to explain that

max_block_generate_distance = 6, max_block_send_distance = 10, viewing_range = 250
![screenshot_20170113_141739](https://cloud.githubusercontent.com/assets/16163058/21931441/b10a5230-d99b-11e6-89ef-4aa0ad23b8dc.png)



max_block_generate_distance = 12, max_block_send_distance = 20, viewing_range = 250
![screenshot_20170113_141953](https://cloud.githubusercontent.com/assets/16163058/21931442/b53722ac-d99b-11e6-8458-e292e6e4d57a.png)

As you can see it doesn't make sense to set the view range to 250 if you don't change other values.

So if you select a preset in the settings menu not only the view range is changed, but also these parameters in minetest.conf with different values for each view range

```
client_unload_unused_data_timeout = 800
client_mapblock_limit = 10000
active_object_send_range_blocks = 5
active_block_range = 5
max_block_send_distance = 15
max_block_generate_distance = 19
```

Maybe the core developer can say more about the values and their effect

**Comment by numberZero** [comment]
Parameter preset should also include various options like bumpmapping, leaves style, etc.
@Fixer-007 That’s what I meant; I though that is mentioned here already.

**Comment by paramat** [comment]
1+ year bump, any core dev support?

**Comment by Ezhh** [comment]
Being able to see more certainly makes the game more attractive, and easier ways for people to get to that are good.

However, mip-mapping is terrible, and I'd consider anything that set that to true for me to be completely unusable.

**Comment by numberZero** [comment]
@Ezhh What’s the problem with mip-mapping?

**Comment by Ezhh** [comment]
@numberZero This is an old with/without mip-mapping screenshot, but unless something major has been changed, should still apply:
![mipmap](https://user-images.githubusercontent.com/7035183/45845916-4ccd6d80-bd1e-11e8-999f-332e494f12c5.png)
It looks pretty terrible. I also feel it cuts in and starts reducing detail much too early. Every time I've used it I've wanted to be able to be able to increase the distance at which it starts doing things. Also see #6846.

**Comment by numberZero** [comment]
Oh, you mean applying mip-mapping to semitransparent textures. That’s problematic at its core. But have you tried enabling `texture_clean_transparent`? It can’t solve #6846 but should be able to fix colors.

**Comment by Ezhh** [comment]
It's much more than the colours. The reduction of texture quality cuts in way too close to the player. Unless there's a way to make it only take effect further away, I think encouraging use of it is not ideal.

**Comment by wsor4035** [comment]
closing this as a dupe of the linked issue as this has had no updates in 7 years and is rather outdated information
